### PR TITLE
Fix parsing of format-items with hollerith items and omitted commas

### PR DESCRIPTION
**Pull request by CoderPuppy**
This fixes parsing of hollerith items in `format-items` in three cases:
- Omitted comma before a hollerith item: `/3Habc`
- Omitted comma after a hollerith item: `3Habc/`
- Trailing whitespace in trailing hollerith item: `4Hspc `

It also changes `Format_Item_List` to match single item lists rather than bail out and rely on `Format_Item`. This matches the behavior of `SequenceBase` and all the generated lists.

### CODE REVIEW

**Comment by arporter** [comment]
Thanks very much for this @CoderPuppy. Are you happy to tackle the missing coverage or would you like someone else to take it from here?

File: src/fparser/two/Fortran2003.py [hunk 1]
```diff
@@ -8881,27 +8903,36 @@ def match(string):
                 if len(current_string) < num_chars:
                     # The string is not long enough.
                     return None
-                item_list.append(Format_Item(current_string[:num_chars]))
+                item_list.append(Hollerith_Item(current_string[:num_chars]))
```
Comment (arporter) [inline src/fparser/two/Fortran2003.py:8906]: Please could you explain the reason for this change?

File: src/fparser/two/tests/fortran2003/test_format_item_list.py [hunk 2]
```diff
@@ -134,6 +134,56 @@ def test_hollerith_only_spaces(f2003_create, monkeypatch):
     )
 
 
+def test_hollerith_omitted_comma_before(f2003_create, monkeypatch):
+    """Check that a hollerith item is parsed correctly when preceeded by a
+    slash or colon without a separating comma.
+
+    """
+    from fparser.two import utils
```
Comment (arporter) [inline src/fparser/two/tests/fortran2003/test_format_item_list.py:142]: Please could you move this import up to the top and replace the use of `format` with f-strings in the new tests (as suggested by pylint).

File: src/fparser/two/tests/fortran2003/test_format_item_list.py [hunk 3]
```diff
@@ -134,6 +134,56 @@ def test_hollerith_only_spaces(f2003_create, monkeypatch):
...
+            "Hollerith_Item('abc')))"
+        ).format(item)
+
+
+def test_hollerith_omitted_comma_after(f2003_create, monkeypatch):
+    """Check that a hollerith item is parsed correctly when followed by a
+    slash or colon without a separating comma.
+
+    """
+    from fparser.two import utils
```
Comment (arporter) [inline src/fparser/two/tests/fortran2003/test_format_item_list.py:160]: Please rm this import.

File: src/fparser/two/tests/fortran2003/test_format_item_list.py [hunk 4]
```diff
@@ -134,6 +134,56 @@ def test_hollerith_only_spaces(f2003_create, monkeypatch):
...
+            "Control_Edit_Desc(None, '{0}')))"
+        ).format(item)
+
+
+def test_hollerith_trailing_space(f2003_create, monkeypatch):
+    """Check that a hollerith item is parsed correctly at the end of a list
+    when it contains a trailing space.
+
+    """
+    from fparser.two import utils
```
Comment (arporter) [inline src/fparser/two/tests/fortran2003/test_format_item_list.py:178]: Please rm this import.

**Review by arporter — CHANGES_REQUESTED** [review]
Changes look good and are well commented (thanks).
I have a question about one changed line and there is some minor tidying to do for pylint in the new tests.
Once that's sorted, this can be merged.
There's no need for a documentation update.

**Comment by CoderPuppy** [comment]
I have a related change to `BracketBase`, preventing the loss of trailing spaces. Should I add that here or create another PR?

File: src/fparser/two/Fortran2003.py (hunk 1 above)
Comment (arporter) [reply src/fparser/two/Fortran2003.py:8906]: Thanks.

**Comment by arporter** [comment]
> I have a related change to BracketBase, preventing the loss of trailing spaces. Should I add that here or create another PR?

A new PR would be great thanks.

**Review by arporter — CHANGES_REQUESTED** [review]
All requested changes have been made.
However, I see CodeCov is reporting a change in coverage which I think is L9094 ( https://app.codecov.io/gh/stfc/fparser/compare/364/changes#D1L9094) in Fortran2003.py (the Format_Item_C1002). Hopefully this just needs a tweak to one of the tests.
Are you OK to take a look at that?

**Comment by CoderPuppy** [comment]
That code is actually redundant now because I implemented slash and comma descriptors in `Format_Item_List`. It can only be triggered by directly calling `Format_Item` or `Format_Item_C1002`. I left it behind to avoid breaking stuff unnecessarily.

It is covered now, but this new test and a few old tests are testing things which shouldn't be able to occur (because standard usage of `Format_Item_C1002` goes through `Format_Item_List`).

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Coverage is now great.
Will proceed to merge.

**Pull request merged**